src/scipp/plt/view.py

```diff
@@ -0,0 +1,156 @@
+            self.fig = self.ax.get_figure()
+
+        self.ax.set_title(title)
+        if grid:
+            self.ax.grid()
+
+        self.left_bar = SideBar([self.toolbar])
+        self.right_bar = SideBar()
+        self.bottom_bar = SideBar()
+        self.top_bar = SideBar()
+
+        self.axformatter = {}
+        self.axlocator = {}
+        self.xlabel = xlabel
+        self.ylabel = ylabel
+        self.draw_no_delay = False
+        self.event_connections = {}
+
+        self.toolbar.connect(view=self)
+
+    def is_widget(self) -> bool:
+        """
+        Check whether we are using the Matplotlib widget backend or not.
+        "on_widget_constructed" is an attribute specific to `ipywidgets`.
+        """
+        return hasattr(self.fig.canvas, "on_widget_constructed")
+
+    def savefig(self, filename: str = None):
+        """
+        Save plot to file.
+        Possible file extensions are `.jpg`, `.png` and `.pdf`.
+        The default directory for writing the file is the same as the
+        directory where the script or notebook is running.
+        """
+        self.fig.savefig(filename, bbox_inches="tight")
+
+    def _ipython_display_(self):
+        """
+        IPython display representation for Jupyter notebooks.
+        """
+        return self._to_widget()._ipython_display_()
+
+    def _to_widget(self) -> ipw.Widget:
+        """
+        Convert the Matplotlib figure to a widget. If the ipympl (widget)
+        backend is in use, return the custom toolbar and the figure canvas.
+        If not, convert the plot to a png image and place inside an ipywidgets
+        Image container.
+        """
+        if self.is_widget():
+            return ipw.VBox([
+                self.top_bar._to_widget(),
+                ipw.HBox([
+                    self.left_bar._to_widget(),
+                    self._to_image() if self.closed else self.fig.canvas,
+                    self.right_bar._to_widget()
+                ]),
+                self.bottom_bar._to_widget()
+            ])
+        else:
+            return self._to_image()
+
+    def _to_image(self) -> ipw.Image:
+        """
+        Convert the Matplotlib figure to a static image.
+        """
+        width, height = self.fig.get_size_inches()
+        dpi = self.fig.get_dpi()
+        return ipw.Image(value=fig_to_pngbytes(self.fig),
+                         width=width * dpi,
+                         height=height * dpi)
+
+    def close(self):
+        """
+        Set the closed flag to True to output static images.
+        """
+        self.closed = True
+
+    def show(self):
+        """
+        Show the matplotlib figure.
+        """
+        self.fig.show()
+
+    def toggle_xaxis_scale(self, change: dict):
+        self.ax.set_xscale("log" if change['new'] else "linear")
+        self.draw()
+
+    def toggle_yaxis_scale(self, change: dict):
+        self.ax.set_yscale("log" if change['new'] else "linear")
+        self.draw()
```

Comment:
Do I read this correctly: The view is matplotlib-specific, and the syntax of `change` is ipywidgets-specific? Is there any reason to couple the two?

Reply:
You're right, i'll remove the ipywidgets specifics